libraries/AP_InertialSensor/BatchSampler.cpp

```diff
@@ -87,8 +87,14 @@ void AP_InertialSensor::BatchSampler::periodic()
 
 void AP_InertialSensor::BatchSampler::update_doing_sensor_rate_logging()
 {
+    if (((batch_opt_t)(_batch_options_mask.get()) & BATCH_OPT_POST_FILTER)
+        && type == IMU_SENSOR_TYPE_GYRO) {
+        _doing_post_filter_logging = true;
```

Comment:
why return here?

Reply:
You can't currently do sensor rate logging with post filter logging because the code paths are different and there is no obvious point to peek at the output. I suppose I could always ignore the sensor rate logging flag if you want post filter, but that seems just as bad as the status quo.

Reply:
See my comment below, the intent is right but there is a bug.